### DISCUSSION TITLE
Allow to change the connected model

### DISCUSSION THREAD

**Discussion by zenzeinet** [TabularEditor/TabularEditor3#1118] [Ideas]
TE3: Allow to change the connected model, without closing all the diagrams, DAX queries, etc! It makes changing models from dev to preprod/prod time consuming

**Comment by otykier** [comment]
Follow-up question: We can certainly leave DAX queries, diagrams and pivot grids open when changing connections, just like we do for C# scripts. However, there are a number of things to be aware of. Please confirm if this behavior aligns with your expectations:

**Diagrams**
If changing connection to a model that does not contain the tables shown in the diagram, these tables will disappear from that diagram. Similarly, if a table exists in the second model, but has different columns or relationships.

**DAX queries**
If the query is not semantically valid on the second model, you will see the red squigglies in the DAX editor, just like when you have any other syntax / semantic error. Moreover, the data results from any previous execution of the query will be cleared.

**Pivot grids**
The pivot grid will remove any fields (columns, measures) that are not present (by name) in the second model.

**Comment by zenzeinet** [comment]
Hi Daniel,

That sounds good - the most important parts are being able to test DAX in low-volume test models, and then switch to preprod/prod for proper testing.

Regarding the behaviours, they are ok. It would be nice if DAX query results could be maintained, but not a huge priority.

Best regards

**Comment by rem-bou** [comment]
I was about to create a similar idea.
It would be a nice addition and very convenient.
